src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -41,222 +43,268 @@
+            valid = false;
+        } else if (!StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.ACCOUNT_NONE, accountDto.getKfsAccount()) 
+                && !StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION, accountDto.getKfsAccount()) 
+                && !validateAccount(accountDto.getKfsChart(), accountDto.getKfsAccount(), logErrorMessage)) {
+            valid = false;
+        } else if (StringUtils.isNotBlank(accountDto.getKfsSubAccount()) 
+                && !validateSubAccountNumber(accountDto.getKfsChart(), accountDto.getKfsAccount(), accountDto.getKfsSubAccount(), logErrorMessage)) {
+            valid = false;
+        } else if (!validateObjectCode(accountDto.getKfsChart(), accountDto.getKfsObject(), logErrorMessage)) {
+            valid = false;
+        } else if (StringUtils.isNotBlank(accountDto.getKfsSubObject()) 
+                && !validateSubObjectCode(accountDto.getKfsChart(), accountDto.getKfsAccount(), accountDto.getKfsObject(), accountDto.getKfsSubObject(), logErrorMessage)) {
+            valid = false;
+        } else if (StringUtils.isNotBlank(accountDto.getKfsProject())  && !validateProjectCode(accountDto.getKfsProject(), logErrorMessage)) {
+            valid = false;
+        } else if (StringUtils.isNotBlank(accountDto.getKfsOrgRefId())  && !validateOrgRefId(accountDto.getKfsOrgRefId(), logErrorMessage)) {
+            valid = false;
+        }
+        return valid;
+    }
+    
+    private AccountingXmlDocumentAccountingLine buildAccountingXmlDocumentAccountingLineFromAmazonKfsAccountDTO(AmazonKfsAccountDTO accountDto) {
+        AccountingXmlDocumentAccountingLine line = new AccountingXmlDocumentAccountingLine();
+        line.setChartCode(accountDto.getKfsChart());
+        line.setAccountNumber(accountDto.getKfsAccount());
+        line.setSubAccountNumber(accountDto.getKfsSubAccount());
+        line.setObjectCode(accountDto.getKfsObject());
+        line.setSubObjectCode(accountDto.getKfsSubObject());
+        line.setProjectCode(accountDto.getKfsProject());
+        line.setOrgRefId(accountDto.getKfsOrgRefId());
+        line.setLineDescription(StringUtils.EMPTY);
+        return line;
+    }
+    
+    private void fixAccountingLine(AccountingXmlDocumentAccountingLine xmlAccountingLine, AmazonKfsAccountDTO defaultAccountDto) {
+        boolean logErrorMessage = true;
+        fixChart(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        fixAccount(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        foxSubAccount(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        foxObjectCode(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        fixSubObjectCode(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        fixProjectCode(xmlAccountingLine, defaultAccountDto, logErrorMessage);
+        foxOrgRefId(xmlAccountingLine, defaultAccountDto, logErrorMessage);
```

Comment:
spelling error

Reply:
I'll change to use the validateAndSetDefaults convention